noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable.nr

```diff
@@ -112,11 +104,17 @@ where
     }
 
     fn read_value_change(self) -> ScheduledValueChange<T> {
-        self.context.storage_read(self.get_value_change_storage_slot())
+        // We don't read ScheduledValueChange directly by having it implement Packable because ScheduledValueChange
+        // and ScheduledDelayChange are packed together (sdc and svc.block_of_change are stored in the same slot).
+        let packed = self.context.storage_read(self.storage_slot);
+        SharedMutableValues::unpack_value_change(packed)
```

Comment:
Doesn't this mean we're reading all slots?

Reply:
I think you mean't `unpack_delay_change`, since values is 2*N+1. But that one accesses `self.storage` slot, which can't be right - you'd need to be reading at `self.storage_slot + 2 * N + 1` to read the single slot where the value. I imagine this means it's being undertested.

And I'm not sure you'll be able to avoid spitting out _some_ layout details if you wan't to make this optimization. Perhaps `SharedMutableValues` should have two fns that return the offset of the value and delay in the storage layout (0 and 2*N + 1).